In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
powers = pd.read_csv('Data/superheroes_power_matrix.csv')
stats = pd.read_csv('Data/charcters_stats.csv')
marvel_characters_info = pd.read_csv('Data/marvel_characters_info.csv')
marvel_DC_characters = pd.read_excel('Data/marvel_dc_characters.xlsx')

In [3]:
powers_df = pd.DataFrame(powers)
stats_df = pd.DataFrame(stats)
marvel_chars_info_df = pd.DataFrame(marvel_characters_info)
marvel_DC_char_df = pd.DataFrame(marvel_DC_characters)

Above I am importing 3 packages
    
    pandas
    numpy
    matplotlib.pyplot
    
Using the pandas package to import the 4 different datasets. From there I again use pandas to create dataframes for each dataset. The datasets can be found on Kaggle: https://www.kaggle.com/dannielr/marvel-superheroes/version/2#marvel_characters_info.csv. 



In [4]:
powers_df.head(10)

Name  Agility  Accelerated Healing  Lantern Power Ring  \
0        3-D Man     True                False               False   
1         A-Bomb    False                 True               False   
2     Abe Sapien     True                 True               False   
3       Abin Sur    False                False                True   
4    Abomination    False                 True               False   
5        Abraxas    False                False               False   
6  Absorbing Man    False                False               False   
7    Adam Monroe    False                 True               False   
8   Adam Strange    False                False               False   
9      Agent Bob    False                False               False   

   Dimensional Awareness  Cold Resistance  Durability  Stealth  \
0                  False            False       False    False   
1                  False            False        True    False   
2                  False             True        True    False   
3                  False            False       False    False   
4                  False            False       False    False   
5                   True            False       False    False   
6                  False             True        True    False   
7                  False            False       False    False   
8                  False            False        True     True   
9                  False            False       False     True   

   Energy Absorption  Flight     ...      Web Creation  Reality Warping  \
0              False   False     ...             False            False   
1              False   False     ...             False            False   
2              False   False     ...             False            False   
3              False   False     ...             False            False   
4              False   False     ...             False            False   
5              False    True     ...             False            False   
6               True   False     ...             False            False   
7              False   False     ...             False            False   
8              False    True     ...             False            False   
9              False   False     ...             False            False   

   Odin Force  Symbiote Costume  Speed Force  Phoenix Force  \
0       False             False        False          False   
1       False             False        False          False   
2       False             False        False          False   
3       False             False        False          False   
4       False             False        False          False   
5       False             False        False          False   
6       False             False        False          False   
7       False             False        False          False   
8       False             False        False          False   
9       False             False        False          False   

   Molecular Dissipation  Vision - Cryo  Omnipresent  Omniscient  
0                  False          False        False       False  
1                  False          False        False       False  
2                  False          False        False       False  
3                  False          False        False       False  
4                  False          False        False       False  
5                  False          False        False       False  
6                  False          False        False       False  
7                  False          False        False       False  
8                  False          False        False       False  
9                  False          False        False       False  

[10 rows x 168 columns]

Shows that the powers dataset seems to be a list of powers that each character might have, denoted on a binary scale.

In [6]:
stats_df.head(25)

Name Alignment  Intelligence  Strength  Speed  Durability  \
0             3-D Man      good            50        31     43          32   
1              A-Bomb      good            38       100     17          80   
2          Abe Sapien      good            88        14     35          42   
3            Abin Sur      good            50        90     53          64   
4         Abomination       bad            63        80     53          90   
5             Abraxas       bad            88       100     83          99   
6         Adam Monroe      good            63        10     12         100   
7        Adam Strange      good             1         1      1           1   
8           Agent  13      good             1         1      1           1   
9           Agent Bob      good            10         8     13           5   
10         Agent Zero      good            75        28     38          80   
11         Air-Walker       bad            50        85    100          85   
12         Alan Scott      good            63        80     23          90   
13       Alex Woolsly      good             1         1      1           1   
14  Alfred Pennyworth      good            63        10     17          10   
15   Allan Quatermain      good             1         1      1           1   
16              Amazo       bad            75       100    100         100   
17               Ammo       bad             1         1      1           1   
18     Ando Masahashi      good             1         1      1           1   
19              Angel      good            63        13     46          64   
20    Angel Salvadore      good            38        10     28          28   
21             Angela       bad             1         1      1           1   
22         Animal Man      good            56        48     47          85   
23          Annihilus       bad            75        80     47          56   
24            Ant-Man      good           100        10     23          28   

    Power  Combat  Total  
0      25      52    233  
1      17      64    316  
2      35      85    299  
3      84      65    406  
4      55      95    436  
5     100      56    526  
6      71      64    320  
7       0       1      5  
8       0       1      5  
9       5      20     61  
10     72      95    388  
11    100      40    460  
12     98      32    386  
13      0       1      5  
14      7      55    162  
15      0       1      5  
16    100     100    575  
17      0       1      5  
18      0       1      5  
19     17      42    245  
20     46      60    210  
21      0       1      5  
22     73      80    389  
23     59      64    381  
24     32      32    225

Looks like there are two scales being used for the data, binary and base 100. For the data that was entered using the a binary scale for the skill measure, the highest possible total is 6 so that threshold is used to split the dataset

In [8]:
stats_adjusted_df_s100 = stats_df[stats_df['Total']>6]
stats_adjusted_df_s_binary = stats_df[stats_df['Total']<7]

In [10]:
stats_adjusted_df_s100.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 434 entries, 0 to 610
Data columns (total 9 columns):
Name            434 non-null object
Alignment       432 non-null object
Intelligence    434 non-null int64
Strength        434 non-null int64
Speed           434 non-null int64
Durability      434 non-null int64
Power           434 non-null int64
Combat          434 non-null int64
Total           434 non-null int64
dtypes: int64(7), object(2)
memory usage: 53.9+ KB


In [11]:
stats_adjusted_df_s_binary.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 177 entries, 7 to 605
Data columns (total 9 columns):
Name            177 non-null object
Alignment       176 non-null object
Intelligence    177 non-null int64
Strength        177 non-null int64
Speed           177 non-null int64
Durability      177 non-null int64
Power           177 non-null int64
Combat          177 non-null int64
Total           177 non-null int64
dtypes: int64(7), object(2)
memory usage: 13.8+ KB


The dataset is split 434 to 177 with the majority being in the 100 scale dataset